src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/ProvisionUtil.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.cloudbreak.service.stack.flow;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Future;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Optional;
+import com.sequenceiq.cloudbreak.controller.BuildStackFailureException;
+import com.sequenceiq.cloudbreak.domain.CloudPlatform;
+import com.sequenceiq.cloudbreak.domain.Stack;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.cloudbreak.repository.StackRepository;
+import com.sequenceiq.cloudbreak.service.stack.resource.ResourceBuilder;
+
+@Component
+public class ProvisionUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProvisionUtil.class);
+
+    @Autowired
+    private StackRepository stackRepository;
+
+    @javax.annotation.Resource
+    private Map<CloudPlatform, List<ResourceBuilder>> instanceResourceBuilders;
+
+    public synchronized boolean isRequestFull(Stack stack, int fullIndex) {
+        return fullIndex % stack.cloudPlatform().parallelNumber() == 0;
+    }
+
+    public synchronized boolean isRequestFullWithCloudPlatform(Stack stack, int fullIndex) {
```

Comment:
A better name would be more helpful as like this it doesn't mean anything. Synchronized is unnecessary in both cases